### PR TITLE
feat: handle bundle yaml and yml files

### PR DESCRIPTION
**Pull request by anthonywendt**
## Description
Updated to handle both uds-bundle.yaml and uds-bundle.yml

## Related Issue

Fixes https://github.com/defenseunicorns/uds-cli/issues/268

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

**Review by decleaver — APPROVED** [review]
pulled in locally and ran some extra checks with the uds-config.yml, looks good!

**Review by decleaver — APPROVED** [review]
pulled in locally and ran some extra checks with the uds-config.yml! 
Reviewed with anthony, looks good

File: src/cmd/uds.go [hunk 1]
```diff
@@ -27,11 +27,20 @@ var createCmd = &cobra.Command{
...
+		if len(args) > 0 {
+			if !zarfUtils.IsDir(args[0]) {
+				message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+			}
+			pathToBundleFile = args[0] + "/"
 		}
-		if _, err := os.Stat(config.BundleYAML); len(args) == 0 && err != nil {
-			message.Fatalf(err, "%s not found in directory", config.BundleYAML)
+		// Handle .yaml or .yml
+		if _, yamlErr := os.Stat(pathToBundleFile + config.BundleYAML); yamlErr == nil {
```
Comment (UncleGedd) [inline src/cmd/uds.go:38]: I see the intent here, but `yamlErr` seems misleading as var name; it's not actually a YAML error

File: src/cmd/uds.go [hunk 2]
```diff
@@ -27,11 +27,20 @@ var createCmd = &cobra.Command{
...
+			bundleCfg.CreateOpts.UserProvidedBundleYamlFile = config.BundleYAML
+		} else if _, ymlErr := os.Stat(pathToBundleFile + strings.Replace(config.BundleYAML, ".yaml", ".yml", 1)); ymlErr == nil {
```
Comment (UncleGedd) [inline src/cmd/uds.go:40]: Referring to `os.Stat(pathToBundleFile + strings.Replace(config.BundleYAML, ".yaml", ".yml", 1))`, can we DRY this up a bit? Repeating the `strings.Replace` bit makes for some long lines

File: src/config/config.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ const (
 	// BlobsDir is the string for the blobs/sha256 dir in an OCI artifact
 	BlobsDir = "blobs/sha256"
 
-	// BundleYAML is the string for zarf.yaml
+	// BundleYAML is the string for uds-bundle.yaml
```
Comment (UncleGedd) [inline src/config/config.go:23]: good catch!

File: src/pkg/bundle/create.go [hunk 4]
```diff
@@ -39,7 +39,7 @@ func (b *Bundler) Create() error {
 	defer os.Chdir(cwd)
 
 	// read the bundle's metadata into memory
-	if err := utils.ReadYaml(config.BundleYAML, &b.bundle); err != nil {
+	if err := utils.ReadYaml(b.cfg.CreateOpts.UserProvidedBundleYamlFile, &b.bundle); err != nil {
```
Comment (UncleGedd) [inline src/pkg/bundle/create.go:42]: `UserProvidedBundleYamlFile` feels a bit verbose as a name. All bundle files are user provided, recommended shortening to something like `BundleFile`

File: src/cmd/uds.go [hunk 5]
```diff
@@ -27,11 +27,20 @@ var createCmd = &cobra.Command{
...
 	Short:   lang.CmdBundleCreateShort,
 	PreRun: func(cmd *cobra.Command, args []string) {
-		if len(args) > 0 && !zarfUtils.IsDir(args[0]) {
-			message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+		pathToBundleFile := ""
+		if len(args) > 0 {
+			if !zarfUtils.IsDir(args[0]) {
+				message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+			}
+			pathToBundleFile = args[0] + "/"
```
Comment (UncleGedd) [inline src/cmd/uds.go:35]: Instead of manually building out paths, we prefer to use `filepath.Join` in Go

File: src/test/e2e/bundle_test.go [hunk 6]
```diff
@@ -272,3 +272,32 @@ func TestBundleWithGitRepo(t *testing.T) {
...
+	zarfPublish(t, pkg, "localhost:889")
+
+	bundleDir := "src/test/bundles/09-uds-bundle-yml"
+	bundlePath := filepath.Join(bundleDir, fmt.Sprintf("uds-bundle-yml-example-%s-0.0.1.tar.zst", e2e.Arch))
+
+	create(t, bundleDir) // todo: allow creating from both the folder containing and direct reference to uds-bundle.yaml
+	inspect(t, bundlePath)
+	inspectAndSBOMExtract(t, bundlePath)
+	// Test with an "options only" config file
+	os.Setenv("UDS_CONFIG", filepath.Join("src/test/bundles/09-uds-bundle-yml", "uds-config.yml"))
```
Comment (UncleGedd) [inline src/test/e2e/bundle_test.go:300]: nice!

**Review by UncleGedd — CHANGES_REQUESTED** [review]

File: src/pkg/bundle/create.go (hunk 4 above)
Comment (anthonywendt) [reply src/pkg/bundle/create.go]: I have irrational flashbacks digging through old fortran code with cryptic single letter variables that I over compensate for. Sorry about that.

File: src/cmd/uds.go (hunk 1 above)
Comment (anthonywendt) [reply src/cmd/uds.go]: Updated

File: src/cmd/uds.go (hunk 2 above)
Comment (anthonywendt) [reply src/cmd/uds.go]: Updated

File: src/cmd/uds.go (hunk 5 above)
Comment (anthonywendt) [reply src/cmd/uds.go]: Updated

File: src/cmd/uds.go [hunk 7]
```diff
@@ -27,11 +27,21 @@ var createCmd = &cobra.Command{
...
+		if len(args) > 0 {
+			if !zarfUtils.IsDir(args[0]) {
+				message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+			}
+			pathToBundleFile = filepath.Join(args[0])
 		}
-		if _, err := os.Stat(config.BundleYAML); len(args) == 0 && err != nil {
-			message.Fatalf(err, "%s not found in directory", config.BundleYAML)
+		// Handle .yaml or .yml
+		bundleYml := strings.Replace(config.BundleYAML, ".yaml", ".yml", 1)
```
Comment (UncleGedd) [inline src/cmd/uds.go:38]: Thanks for that refactor. One more thing, looks like you can remove the `pathToBundleFile` variable entirely and replace it in those `if` statements with `args[0]`

Comment (anthonywendt) [reply src/cmd/uds.go:38]: I don't believe I can do that. The way I wrote this code, the handle .yaml or .yml section can handle either file type, whether or not the user provided a directory to the create command. If I make the change you suggested and the user doesn't provide a directory, args[0] will error with index out of range. This is why, while checking if the user provided a directory and checking if its a valid directory I join it to the `pathToBundleFile` string. Using the `pathToBundleFile` string allows the following .yaml vs .yml checks to work in either situation without issue. Unless I'm overlooking something or misunderstanding the suggestion.

Comment (UncleGedd) [reply src/cmd/uds.go:38]: You are def correct, thanks for that clarification!

**Review by UncleGedd — APPROVED** [review]

**Pull request merged**